bitcoin/src/psbt/map/global.rs

```diff
@@ -35,10 +35,10 @@ impl Map for PartiallySignedTransaction {
                 // Manually serialized to ensure 0-input txs are serialized
                 // without witnesses.
                 let mut ret = Vec::new();
-                self.unsigned_tx.version.consensus_encode(&mut ret)?;
-                self.unsigned_tx.input.consensus_encode(&mut ret)?;
-                self.unsigned_tx.output.consensus_encode(&mut ret)?;
-                self.unsigned_tx.lock_time.consensus_encode(&mut ret)?;
+                ret.extend(encode::serialize(&self.unsigned_tx.version));
```

Comment:
These should be psbt::Serialize, not encode::Serialize.

Reply:
Should they be? None of `version: i32`, `input`, `output`, `lock_time` are `psbt::map::Serialize` yet.

BIP 174 spec PSBT_GLOBAL_UNSIGNED_TX <valuedata> = "The transaction in [consensus] network serialization. The scriptSigs and witnesses for each input must be empty. The transaction must be in the old serialization format (without witnesses)."

Fortunately PSBTv2 does away with this hack that nests the consensus raw tx map within the psbt map and gives everything relevant in psbt a proper field.